### PR TITLE
feat: interpret Intfunction as Int32 -> Object

### PR DESCRIPTION
**Pull request by mlutze**
related to #4721

### CODE REVIEW

**Comment by magnus-madsen** [comment]
It should be possible to write some tests that result in ClassCastException (until the backend is fixed).

Perhaps adding all the different functions could be done by a student programmer? (If there is a "template" in place).

**Comment by mlutze** [comment]
> It should be possible to write some tests that result in ClassCastException (until the backend is fixed).

Wouldn't those tests be better written by the backend developer? :P

> Perhaps adding all the different functions could be done by a student programmer? (If there is a "template" in place).

Yes, the pattern I've used here ought to work for the rest of the function types.

**Comment by magnus-madsen** [comment]
I think the test actually needs to pass the function into something in Java. E.g. something with streams.

**Comment by magnus-madsen** [comment]
> Wouldn't those tests be better written by the backend developer? :P

I think there should be tests written by both, no? :)

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala [hunk 1]
```diff
@@ -2653,21 +2657,31 @@ object Resolver {
...
         case TypeConstructor.Union => ResolutionError.IllegalType(tpe, loc).toFailure
 
+        case t: TypeConstructor.Arrow => throw InternalCompilerException(s"unexpected type: $t")
+        case t: TypeConstructor.Enum => throw InternalCompilerException(s"unexpected type: $t")
 
       }
 
-      // Case 2: Enum. Return an object type.
+      // Case 2: Arrow. Convert to Java function interface
+      case UnkindedType.Arrow(_, _, _) =>
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:2666]: What's being converted here? Why are we converting TO IntFunction?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:2666]: Makes sense 👍

File: main/src/ca/uwaterloo/flix/language/phase/Resolver.scala [hunk 2]
```diff
@@ -1853,6 +1853,12 @@ object Resolver {
       fqn match {
         case "java.math.BigInteger" => UnkindedType.Cst(TypeConstructor.BigInt, loc).toSuccess
         case "java.lang.String" => UnkindedType.Cst(TypeConstructor.Str, loc).toSuccess
+        case "java.util.function.IntFunction" =>
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:1856]: Could this be one line, since there is going to be many lines?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Resolver.scala:1856]: Perhaps these constants should be elsewhere?

**Comment by magnus-madsen** [comment]
Is this ready to be merged and handed-off to Jakob and Jonathan?

**Comment by mlutze** [comment]
> Is this ready to be merged and handed-off to Jakob and Jonathan?

Yes

**Comment by magnus-madsen** [comment]
@JonathanStarup Matt did something for the frontend.

**Pull request merged**